sdk/identity/azure-identity/src/azure_pipelines_credential.cpp

```diff
@@ -101,18 +73,10 @@ AzurePipelinesCredential::AzurePipelinesCredential(
             + "' needed by " + GetCredentialName() + ". This should be set by Azure Pipelines.");
   }
 
-  if (isTenantIdValid && !clientId.empty() && !serviceConnectionId.empty()
-      && !systemAccessToken.empty() && !m_oidcRequestUrl.empty())
+  m_credentialCreatedSuccessfully = TenantIdResolver::IsValidTenantId(tenantId) && !clientId.empty()
+      && !serviceConnectionId.empty() && !systemAccessToken.empty() && !m_oidcRequestUrl.empty();
+  if (m_credentialCreatedSuccessfully)
```

Comment:
The naming of the variable might be confusion, but `m_credentialCreatedSuccessfully` is talking about successful construction of AzurePipelinesCredential (which requires more checks) and not ClientAssertionCredential.

Just because ClientAssertionCredential got created doesn't mean AzurePipelinesCredential is valid, since we still need to check the other params such as service connection id and system access token. Only if those input params are good, do we set `m_credentialCreatedSuccessfully`.

But I will see if we can remove the bool using the `reset` approach.

Reply:
If you move the check in the ctor above the code which creates the credential, what other aspects of AzurePipelineCredentials construction would necessitate the addition of the additional variable.

Reply:
That's a good point. I can create the credential only in the case where the validation conditions are met (thus not even needing to reset it in the else case).